cg/meta/upload/gisaid/gisaid.py

```diff
@@ -74,131 +107,152 @@ def get_gisaid_samples(self, case_id: str) -> List[GisaidSample]:
-        fasta_file: File = self.housekeeper_api.find_file_in_latest_version(
-            case_id=case_id, tags=["consensus"]
-        )
-
-        gisaid_delivery_fasta = []
-        with open(str(fasta_file.full_path)) as handle:
-            fasta_lines = handle.readlines()
-            for line in fasta_lines:
-                if line[0] == ">":
-                    gisaid_delivery_fasta.append(
-                        self.get_new_header(old_header=line, samples=gisaid_samples)
-                    )
-                else:
-                    gisaid_delivery_fasta.append(line)
-        if not gisaid_delivery_fasta:
-            raise InvalidFastaError(message="Empty fasta file")
-        return gisaid_delivery_fasta
-
-    def get_new_header(self, samples: List[GisaidSample], old_header: str) -> str:
-        for sample in samples:
-            sample_id = sample.covv_subm_sample_id
-            if old_header.find(sample_id) != -1:
-                return f">{sample.covv_virus_name}\n"
-        raise FastaSequenceMissingError
-
-    def build_gisaid_fasta(
-        self, gisaid_samples: List[GisaidSample], file_name: str, case_id: str
-    ) -> Path:
+    def create_gisaid_fasta(self, gisaid_samples: List[GisaidSample], case_id: str) -> None:
         """Writing a new fasta with headers adjusted for gisaid upload_results_to_gisaid"""
 
-        file: Path = Path(file_name)
-        fasta_lines: List[str] = self.get_gisaid_fasta(
-            gisaid_samples=gisaid_samples, case_id=case_id
+        gisaid_fasta_file = self.housekeeper_api.find_file_in_latest_version(
+            case_id=case_id, tags=["gisaid-fasta", case_id]
         )
+        if gisaid_fasta_file:
+            gisaid_fasta_path = gisaid_fasta_file.full_path
+        else:
+            gisaid_fasta_path: Path = self.get_gisaid_fasta_path(case_id=case_id)
 
-        with open(file, "w") as write_file_obj:
-            write_file_obj.writelines(fasta_lines)
-        return file
-
-    def get_sample_row(self, gisaid_sample: GisaidSample) -> List[str]:
-        """Build row for a sample in the batch upload_results_to_gisaid csv."""
-
-        sample_dict = gisaid_sample.dict()
-        return [sample_dict.get(header, "") for header in HEADERS]
-
-    def get_sample_rows(self, gisaid_samples: List[GisaidSample]) -> List[List[str]]:
-        """Build sample row list for gisad csv file"""
+        fasta_lines: List[str] = []
 
-        sample_rows = []
         for sample in gisaid_samples:
-            sample_row: List[str] = self.get_sample_row(gisaid_sample=sample)
-            sample_rows.append(sample_row)
-        return sample_rows
+            fasta_file: File = self.housekeeper_api.find_file_in_latest_version(
+                case_id=case_id, tags=[sample.cg_lims_id, "consensus-sample"]
+            )
+            if not fasta_file:
+                LOG.warning(f"No fasta file found for sample {sample.cg_lims_id}")
```

Comment:
I forgot... Why would we continue if a fasta file is missing? Is this handeled when building the csv file? Or will there be a missmach between fasta and csv then? Will gisaid acceppt that?

Reply:
good point